packages/tse-backend/src/modules/exchange-data/exchange-data.service.ts

```diff
@@ -0,0 +1,298 @@
+  }
+
+  async getMultipleTickerPrices(command: GetMultipleTickerPricesCommand) {
+    const { exchangeNames, symbols } = command;
+    const fetchPromises: Promise<TickerPriceResponse>[] = [];
+    const results: {
+      [exchange: string]: { [symbol: string]: TickerPriceResponse };
+    } = {};
+
+    const exchangeNamesArray = exchangeNames.split(',');
+    const symbolsArray = symbols.split(',');
+
+    exchangeNamesArray.forEach((exchangeName) => {
+      symbolsArray.forEach((symbol) => {
+        const tpCommand = new GetTickerPriceCommand(exchangeName, symbol);
+        const fetchPromise = this.getTickerPrice(tpCommand)
+          .then((price) => {
+            if (!results[exchangeName]) {
+              results[exchangeName] = {};
+            }
+            results[exchangeName][symbol] = price;
+            return price;
+          })
+          .catch((error) => {
+            this.logger.error(
+              `Error fetching ticker price for ${exchangeName}:${symbol}: ${error}`,
+            );
+            return null;
+          });
+        fetchPromises.push(fetchPromise);
+      });
+    });
+
+    await Promise.all(fetchPromises);
+    return results;
+  }
+
+  async getSupportedSymbols(command: GetSupportedSymbolsCommand) {
+    const { exchange } = command;
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance) {
+      throw new Error(`Exchange ${exchange} is not configured.`);
+    }
+    this.logger.log(`Fetching supported symbols from ${exchangeInstance.name}`);
+
+    try {
+      await exchangeInstance.loadMarkets();
+      return Object.keys(exchangeInstance.markets);
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  private async watchMarketData(
+    type: MarketDataType,
+    exchange: string,
+    symbol: string | undefined,
+    symbols: string[] | undefined,
+    timeFrame: string | undefined,
+    onData: (data: any) => void,
+    extraParams: any = {},
+  ): Promise<void> {
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    const methodName = `watch${type}`;
+    const compositeKey = createCompositeKey(
+      type,
+      exchange,
+      symbol,
+      symbols,
+      timeFrame,
+    );
+
+    if (!exchangeInstance || !exchangeInstance.has[methodName]) {
+      throw new Error(
+        `Exchange ${exchange} does not support ${methodName} or is not configured.`,
+      );
+    }
+
+    try {
+      while (this.subscriptionManager.isSubscribed(compositeKey)) {
+        const data = await exchangeInstance[methodName](
+          symbol || symbols,
+          ...Object.values(extraParams),
+        );
+        onData(data);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error watching ${type} for ${symbol || symbols} on ${exchange}: ${error.message}`,
+      );
+      await new Promise((resolve) => setTimeout(resolve, 1000)); // Reconnect after a delay
```

Comment:
What's the purpose of that?

Reply:
In cases where the ccxt API is temporarily unavailable, we don't want to exit the loop; instead, we want to keep receiving data. This delay helps us avoid sending too many requests. However, since they don't use this delay in their example (https://github.com/ccxt/ccxt/blob/master/examples/ccxt.pro/js/gateio-watch-order-book.js#L31), and it may not be necessary now, I'll remove this line.

Reply:
fix: 1fb1dc4e

Reply:
Did you test it, if it's not necessary?

Reply:
It will be hard to test this. I checked the forums and didn't find anyone who had a problem with the ban. As for exchange restrictions, CCXT has a built-in rate limiter that automatically manages the request speed.